src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 8;
+    private T[] storage;
+    private int size;
+
+    public ArrayList() {
+        storage = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        checkLength();
+        storage[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index >= 0 && index <= size) {
+            checkLength();
+            System.arraycopy(storage, index, storage,index + 1, size - index);
+            storage[index] = value;
+            size++;
+        } else {
+            checkIndex(index);
+        }
```

Comment:
```suggestion
        if (index < 0 || index > size) {
            throw new ArrayListIndexOutOfBoundsException("Index: " + index + ", Size: " + size);
        }
        checkLength();
        System.arraycopy(storage, index, storage,index + 1, size - index);
        storage[index] = value;
        size++;
```
In this case we have a different check from checkIndex(), so we don't need this method here at all.